src/jobs/lint-app.yml

```diff
@@ -1,9 +1,92 @@
-description: >
+description: |
   This job will lint the application based on what language the application is written in.
 
-executor: node/default
+parameters:
+  # Not Required Parameters
+  docker-image:
+    type: string
+    default: "cimg/base:stable"
+    description: |
+      The docker image that will be used as an executor
+      if isNode is true, node/default is recommended
+      if isGolang is true, golangci/golangci-lint:latest-alpine is recommended
+      if isPython is true, cimg/python:3.10 is recommended
+  isNode:
+    type: boolean
+    default: false
+    description: |
+      Set `true` if wanting to lint a node.js application
+      Will use the script `npm run lint`
+  isGolang:
+    type: boolean
+    default: false
+    description: |
+      Set `true` if wanting to lint a Golang application
+      Will use the golangci-lint project
+  isPython:
+    type: boolean
+    default: false
+    description: |
+      Set `true` if wanting to lint a Python application
+      Will use flake8 and pylint
+  golang-directory:
+    type: string
+    default: ./
+    description: |
+      Set the directory where the linter should run.
+      (This only applies if isGolang is true)
+  python-directory:
+    type: string
+    default: src
+    description: |
+      Set the directory where the linter should run.
+      (This only applies if isPython is true)
+
+docker:
+  - image: <<parameters.docker-image>>
 
 steps:
+  - checkout
```

Comment:
Is there a reason why this example checks out the repository twice?

Reply:
Good catch